src/Diff.re

```diff
@@ -10,55 +10,96 @@ module MakeDiff = (IO1: ImageIO.ImageIO, IO2: ImageIO.ImageIO) => {
       (
         base: ImageIO.img(IO1.t),
         comp: ImageIO.img(IO2.t),
+        ~antialiasing=false,
         ~outputDiffMask=false,
         ~diffPixel: (int, int, int)=redPixel,
         ~threshold=0.1,
         (),
       ) => {
-    let diffCount = ref(0);
+    let diffPixels = ref([]);
     let maxDelta = maxYIQPossibleDelta *. threshold ** 2.;
     let diffOutput = outputDiffMask ? IO1.makeSameAsLayout(base) : base;
 
     let countDifference = (x, y) => {
-      diffCount := diffCount^ + 1;
-      diffOutput |> IO1.setImgColor(x, y, diffPixel);
+      diffPixels := [(x, y), ...diffPixels^];
     };
 
     for (y in 0 to base.height - 1) {
       let row = IO1.readRow(base, y);
       let row2 = IO2.readRow(comp, y);
 
+      let getBaseColor = (x, rowToGet) => {
+        IO1.readImgColor(
+          x,
+          rowToGet == y ? row : IO1.readRow(base, rowToGet),
+          base,
+        );
+      };
+
+      let getCompColor = (x, rowToGet) => {
+        IO2.readImgColor(
+          x,
+          rowToGet == y ? row2 : IO2.readRow(comp, rowToGet),
+          comp,
+        );
+      };
+
       for (x in 0 to base.width - 1) {
         if (x >= comp.width || y >= comp.height) {
           let (_r, _g, _b, a) = IO1.readImgColor(x, row, base);
           if (a != 0) {
             countDifference(x, y);
           };
         } else {
-          let (r, g, b, a) = IO1.readImgColor(x, row, base);
-          let (r1, g1, b1, a1) = IO2.readImgColor(x, row2, comp);
+          let baseColor = IO1.readImgColor(x, row, base);
+          let compColor = IO2.readImgColor(x, row2, comp);
 
-          if (r != r1 || g != g1 || b != b1 || a != a1) {
+          if (!Helpers.isSameColor(baseColor, compColor)) {
             let delta =
-              ColorDelta.calculatePixelColorDelta(
-                (r, g, b, a),
-                (r1, g1, b1, a1),
-              );
+              ColorDelta.calculatePixelColorDelta(baseColor, compColor);
 
             if (delta > maxDelta) {
-              countDifference(x, y);
+              let isAntialiased =
+                if (!antialiasing) {
+                  false;
+                } else {
+                  Antialiasing.isAntialiased(
+                    ~x,
+                    ~y,
+                    ~width=base.width,
+                    ~height=base.height,
+                    ~readBaseColor=getBaseColor,
+                    ~readCompColor=getCompColor,
```

Comment:
We are having an issue here, when the base image is taller (or wider) than the comp image.
When we check for antialiasing near the edge, we could go out of bounds.

TODO: Add the compWidth / compHeight to the function and clip the neighbouring pixel check to the correct dimensions.

(Just leaving this here, so I don't forget)